packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -951,6 +970,36 @@ class _ToggleButton extends StatelessWidget {
   }
 }
 
+@immutable
+class _ResolveFillColor extends MaterialStateProperty<Color?> with Diagnosticable {
+  _ResolveFillColor(this.primary);
+
+  final Color? primary;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (primary is MaterialStateProperty<Color>) {
+      return MaterialStateProperty.resolveAs<Color?>(primary, states);
+    }
+    return states.contains(MaterialState.selected) ? primary : null;
```

Comment:
The default value should be handled by _DefaultColor. This class isn't needed.

Reply:
This class doesn't handle default value but handles when to resolve the color or what value should return if its property is color.